packages/vitest/src/node/core.ts

```diff
@@ -386,6 +381,11 @@ export class Vitest {
   async start(filters?: string[]) {
     this._onClose = []
 
+    // if Vitest is running globally, then we should still import local vitest if possible
+    const projectVitestPath = await this.vitenode.resolveId('vitest')
+    const vitestDir = projectVitestPath ? resolve(projectVitestPath.id, '../..') : rootDir
```

Comment:
vs code extension doesn't call `start` at all

Reply:
Oh, really...
I had `createRequire` approach before https://github.com/vitest-dev/vitest/pull/5646/commits/10375119948a856ec62ac1adf73ea140cab27e8c. Would this work better?

Reply:
Or we could delay it further to `ctx.runFiles`. Let me think...

Reply:
The `createRequire` approach is wrong because we need Vite to process it. Having it in runFiles is fine. We can have a guard to not check this path if it is resolved, and reset it in `configureServer`

Reply:
Do we need Vite's resolution? If the purpose is only to find local Vitest, would `createRequire` with `require.resolve('vitest', { paths: [config.root] })` do the same?

This was the approach used for package installer:

https://github.com/vitest-dev/vitest/blob/04df286ee3e88db328148122face9e4a01fee084/packages/vitest/src/node/packageInstaller.ts#L15-L18

Reply:
We need Vite's resolution because files importing vitest are processed by Vite. It's possible that Vitest is called globally and resolved id is not correct, or there is an alias or resolveId interception